components/dialog/dialog-mixin.js

```diff
@@ -3,7 +3,7 @@ import '../../helpers/viewport-size.js';
 import { allowBodyScroll, preventBodyScroll } from '../backdrop/backdrop.js';
 import { clearDismissible, setDismissible } from '../../helpers/dismissible.js';
 import { findComposedAncestor, isComposedAncestor } from '../../helpers/dom.js';
-import { forceFocusVisible, getComposedActiveElement, getNextFocusable, isFocusVisibleApplied, tryApplyFocus } from '../../helpers/focus.js';
+import { getComposedActiveElement, getFirstFocusableDescendant, getNextFocusable, isFocusable, isFocusVisibleApplied } from '../../helpers/focus.js';
```

Comment:
Do you know if the `isFocusVisibleApplied` usage still needed?

Reply:
Hmm. So when the dialog opens, it's still using `isFocusVisibleApplied` to check if the active element has focus-visible applied and if not, it makes another attempt at finding and focusing. That [was introduced here](https://github.com/BrightspaceUI/core/commit/a1eb5d3d1d162d27d0b161a8908eff30166c4fb8).

This is a little weird now though, since clicking with the mouse is going to result in that always being `false`, compared to using the keyboard.

IMO, the check on the line above that ensures the active element is _inside_ the dialog should be enough to convince us it's OK to focus on it?

Reply:
That's what I'ma thinking too.

Reply:
Yeah I was hoping `isFocusVisibleApplied` could be removed with these changes, but I didn't fully understand the motivation behind it. It's also used in `animate.js` btw.